A72-open-telemetry-tracing.md

```diff
@@ -0,0 +1,630 @@
+  // Set one or multiple text map propagators for span context propagation, e.g.
+  // the community standard ones like W3C, etc.
+  OpenTelemetryPluginBuilder& SetTextMapPropagator(
+      std::unique_ptr<opentelemetry::context::propagation::TextMapPropagator>
+          text_map_propagator);
+};
+```
+
+### Python
+The following new fields will be added in `OpenTelemetryPlugin`.
+
+```Python
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.trace.propagation.tracecontext import TraceContextTextMapPropagator
+
+class OpenTelemetryPlugin:
+    """
+    If `tracer_provider` is None, no traces are collected.
+    """
+
+    def __init__(
+        self,
+        *,
+        tracer_provider: Optional[TracerProvider] = None,
+        text_map_propagator: [TraceContextTextMapPropagator] = None,
+    ):
+```
+
+### Go
+The following `TraceOptions` field will be added to the `opentelemetry` Options struct.
+```go
+import (
+  "go.opentelemetry.io/otel/trace"
+)
+
+type Options struct {
+  // Existing field in opentelemetry package
+  MetricsOptions MetricsOptions
+  TraceOptions TraceOptions
+}
+
+// TraceOptions are the trace options for OpenTelemetry instrumentation.
+type TraceOptions struct {
+  TraceProvider trace.TraceProvider
+}
+
+```
+
+## Tracing Information
+RPCs on the client side may undergo retry attempts, whereas on the server side, 
+they do not. gRPC records both per-call tracing details (on the parent span) 
+and per-attempt tracing details (on the attempt span) on the client side. 
+On the server side, there is only per-call traces.  With the new OpenTelemetry 
+plugin we will produce the following tracing information during an RPC lifecycle:
+
+At the client, on parent span:
+* If the RPC experienced name resolution delay, add an Event at the start of the
+  call with the name "Delayed name resolution complete" upon completion of the
+  name resolution process.
+* When the call is closed, set RPC status and end the parent span. gRPC status "OK" 
+  is recorded with status "OK", while other gRPC statuses are marked as "ERROR". 
+  Non-"OK" statuses include their code as a description, the same below.
+
+On attempt span:
+* When span is created, set the attribute with key `previous-rpc-attempts` and an 
+  integer value representing the count of previous attempts made for the RPC.
+* When span is created, set the attribute with key `transparent-retry` and a 
+  boolean value indicating whether the stream is undergoing a transparent retry.
+* If the RPC experienced load balancer pick delay, add an Event with the name 
+  "Delayed LB pick complete" upon creation of the stream on the transport. 
+* When the application sends an outbound message, add Event(s)
+  (it depends on implementation whether there is a single event or an additional 
+  separate event with name "Outbound message compressed" for compressed message size) 
+  with name "Outbound message sent" and the following attributes:
```

Comment:
I think we talked about this earlier, but "sent" in "Outbound message sent" is confusing. Can we drop it?
For C++, this would otherwise mean that we would first see - 
"Outbound message sent" and then "Outbound message compressed"

Reply:
> "Outbound message sent" and then "Outbound message compressed"

It looks fine. "sent" means this trace happens somewhere from application to the wire, and we made that vague and general(decision made from a discussion). Are you confused because compression should not happen after "sent"?